radicle-cob/src/test/crypto.rs

```diff
@@ -0,0 +1,229 @@
+        buf[..2].copy_from_slice(&Self::MULTICODEC_TYPE);
+        buf[2..].copy_from_slice(self.0.deref());
+
+        multibase::encode(multibase::Base::Base58Btc, &buf)
+    }
+}
+
+/// Cryptographic signature.
+#[derive(PartialEq, Eq, Copy, Clone, Serialize, Deserialize)]
+#[serde(into = "String", try_from = "String")]
+pub struct Signature(pub ed25519::Signature);
+
+impl fmt::Display for Signature {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let base = multibase::Base::Base58Btc;
+        write!(f, "{}", multibase::encode(base, self.deref()))
+    }
+}
+
+impl fmt::Debug for Signature {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Signature({})", self)
+    }
+}
+
+#[derive(Error, Debug)]
+pub enum SignatureError {
+    #[error("invalid multibase string: {0}")]
+    Multibase(#[from] multibase::Error),
+    #[error("invalid signature: {0}")]
+    Invalid(#[from] ed25519::Error),
+}
+
+impl From<ed25519::Signature> for Signature {
+    fn from(other: ed25519::Signature) -> Self {
+        Self(other)
+    }
+}
+
+impl FromStr for Signature {
+    type Err = SignatureError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let (_, bytes) = multibase::decode(s)?;
+        let sig = ed25519::Signature::from_slice(bytes.as_slice())?;
+
+        Ok(Self(sig))
+    }
+}
+
+impl Deref for Signature {
+    type Target = ed25519::Signature;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl From<[u8; 64]> for Signature {
+    fn from(bytes: [u8; 64]) -> Self {
+        Self(ed25519::Signature::new(bytes))
+    }
+}
+
+impl TryFrom<&[u8]> for Signature {
+    type Error = ed25519::Error;
+
+    fn try_from(bytes: &[u8]) -> Result<Self, Self::Error> {
+        ed25519::Signature::from_slice(bytes).map(Self)
+    }
+}
+
+impl TryFrom<String> for Signature {
+    type Error = SignatureError;
+
+    fn try_from(s: String) -> Result<Self, Self::Error> {
+        Self::from_str(&s)
+    }
+}
+
+impl From<Signature> for String {
+    fn from(sig: Signature) -> Self {
+        sig.to_string()
+    }
+}
+
+impl AsRef<[u8]> for Signature {
+    fn as_ref(&self) -> &[u8] {
+        self.0.as_ref()
+    }
+}
+
+impl sign::PublicKey for PublicKey {
+    type Signature = Signature;
+
+    fn verify(&self, signature: &Self::Signature, bytes: &[u8]) -> bool {
+        self.deref().verify(bytes, signature).is_ok()
+    }
+}
+
+/// The private/signing key.
+pub type SecretKey = ed25519::SecretKey;
+
+pub fn secret_key() -> SecretKey {
+    ed25519::KeyPair::generate().sk
+}
+
+impl sign::Signer for SecretKey {
+    type Error = Infallible;
+    type Signature = Signature;
+    type Key = PublicKey;
+
+    fn sign(&self, bytes: &[u8]) -> Result<Self::Signature, Self::Error> {
+        Ok(Signature(ed25519::SecretKey::sign(self, bytes, None)))
+    }
+
+    fn public_key(&self) -> Self::Key {
+        PublicKey(ed25519::SecretKey::public_key(self))
+    }
+}
```

Comment:
All this would also go away with a `radicle-crypto` crate

Reply:
Yes, well I'm glad we got on the same page. I'll definitely move it out :joy: